includes/class-wcpdf-main.php

```diff
@@ -588,6 +590,34 @@ public function init_tmp () {
 		}
 	}
 
+	public function load_custom_and_local_fonts() {
+		require WPO_WCPDF()->plugin_path() . '/vendor/autoload.php';
+
+		$options = array(
+			'defaultFont'				=> 'dejavu sans',
+			'tempDir'					=> $this->get_tmp_path( 'dompdf' ),
+			'logOutputFile'				=> $this->get_tmp_path( 'dompdf' ) . "/log.htm",
+			'fontDir'					=> $this->get_custom_fonts_path(),
+			'fontCache'					=> $this->get_custom_fonts_path(),
+			'isRemoteEnabled'			=> true,
+			'isFontSubsettingEnabled'	=> true,
+			'isHtml5ParserEnabled'		=> true,
+		);
+
+		// save custom fonts
+		$dompdf = new Dompdf( new Options( $options ) );
+		$dompdf->getFontMetrics()->saveFontFamilies();
+
+		// save local fonts
+		$options['fontDir'] = $options['fontCache'] = $this->get_tmp_path( 'fonts' );
```

Comment:
would this result in an infinite loop if `load_custom_and_local_fonts()` is called by `init_tmp()`? The goal is to make this method replace the current `copy_fonts()` method.

Reply:
Well, `copy_fonts` calls `get_tmp_path` also: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/2ebe7cb9be7248bfe1cb7a4ea81cb6037ac7404a/includes/class-wcpdf-main.php#L696-L697

Reply:
I see, so it's probably safe, but something to keep in mind.